subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/ConfigurationCacheHost.kt

```diff
@@ -178,11 +177,7 @@ class ConfigurationCacheHost internal constructor(
         fun createSettings(): SettingsState {
             val baseClassLoaderScope = gradle.classLoaderScope
             val classLoaderScope = baseClassLoaderScope.createChild("settings", null)
-            val settingsSource = if (settingsFile == null) {
-                TextResourceScriptSource(StringTextResource("settings", ""))
-            } else {
-                TextResourceScriptSource(service<TextFileResourceLoader>().loadFile("settings file", settingsFile))
-            }
+            val settingsSource = TextResourceScriptSource(service<TextFileResourceLoader>().loadFile("settings file", settingsFile))
```

Comment:
Ensuring we use the same code as elsewhere to obtain a Text Resource script source